languages/js/src/helpers.ts

```diff
@@ -106,19 +102,20 @@ function parseResult({ bindings }: obj): QueryEvent {
  *
  * @internal
  */
-function parseMakeExternal(d: obj): QueryEvent {
-  const instanceId = d.instance_id;
-  const ctor = d['constructor']?.value?.Call;
-  const tag = ctor?.name;
-  const fields = ctor?.args;
-  if (ctor?.kwargs) throw new KwargsError();
-  if (
-    !Number.isSafeInteger(instanceId) ||
-    typeof tag !== 'string' ||
-    !Array.isArray(fields) ||
-    fields.some((v: unknown) => !isPolarTerm(v))
-  )
-    throw new Error();
+function parseMakeExternal(event: unknown): QueryEvent {
+  if (!isObj(event)) throw new Error();
+  const { instance_id: instanceId } = event;
+  if (!isSafeInteger(instanceId)) throw new Error();
+  const ctor = event['constructor'];
+  if (!isPolarTerm(ctor)) throw new Error();
+  if (!isPolarPredicate(ctor.value)) throw new Error();
```

Comment:
there's so many of these I feel like we should just write `assert()`

Reply:
https://github.com/osohq/oso/pull/1211#discussion_r714507826